runtime/src/network/iouring.rs

```diff
@@ -0,0 +1,207 @@
+use crate::iouring;
+use commonware_utils::{StableBuf, StableBufMut};
+use futures::{
+    channel::{mpsc, oneshot},
+    executor::block_on,
+    SinkExt as _,
+};
+use io_uring::{squeue::Entry as SqueueEntry, types::Fd};
+use std::{
+    net::SocketAddr,
+    os::fd::{AsRawFd, OwnedFd},
+    sync::Arc,
+};
+use tokio::net::{TcpListener, TcpStream};
+
+#[derive(Clone, Debug)]
+/// [crate::Network] implementation that uses io_uring to do async I/O.
+pub struct Network {
+    submitter: mpsc::Sender<(SqueueEntry, oneshot::Sender<i32>)>,
+}
+
+impl Network {
+    /// Returns a new [Network] instance.
+    /// This function starts a new thread to run the io_uring event loop.
+    /// The thread will run until the work submission channel is closed or
+    /// the event loop errors.
+    pub(crate) fn start(cfg: iouring::Config) -> Result<Self, crate::Error> {
+        let (tx, rx) = mpsc::channel(128);
```

Comment:
We should make this a config-driven (like we do with storage): https://github.com/commonwarexyz/monorepo/blob/c47e036969ec793c8d5057a4f1d4a8cf44066e64/runtime/src/storage/iouring.rs#L34

Reply:
🤦 fixed